module-2/backend/week_two.md

```diff
@@ -5,33 +5,77 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+* Student     Enrollment    Teacher
+* id           id            id
+*              teacher_id   
+*              student_id
+
+* student has many teachers. teachers have many students
+
 8. Define foreign key, primary key, and schema.
+* Foreign key - primary key of a second table associated by the primary key of a first table
+* Primay key - column(s) that uniquely identify a record
+* Schema (db) - overarching organization and structure of database
+* Schema (file) - displays all entities and their structure
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+* The primary key points to the foriegn key. Primary key cannot be null or duplicate, foreign key can. Primary key must be a unique key, foreign key may or may not uniquely identify a row
 10. What are the parts of an HTTP response?
+* status, headers, body
 11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+* use partial views, use helper methods, use modules
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+* delete_all - truncates table
+* pluck - select column
+* distinct - unique records
+* group - group records
+* order - sort records
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+* rake db:reset - resets and reloads db
+* rake db:schema:load - loads the schema 
+* rake db:test:prepare - prepares db for rspec
 4. What can you expect from a group as you begin working together? As you continue working together?
+* expect people to have different working habits
+* expect people to contribute and compromise
 5. What two columns does `t.timestamps null: false` create in our database?
+* created_at, update_at
 6. What cURL flag can you use to send a `POST` request?
+* -d 
 7. What case does JSON (and JavaScript) use for multi-word variables?
+* camelcase
 8. What case does Ruby use for multi-word variables?
+* snake case
 9. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+* school has many teachers, teachers (usually) belong to one school
 10. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+* Teachers will have a school id
 11. Give an example of when you might want to store information besides ids on a join table.
+* join name
 12. Describe and diagram the relationship between patients and doctors.
+* usually, patients have one doctor, doctors have many patients
 13. Describe and diagram the relationship between museums and original_paintings.
+* museums have many og paintings, paintings belong to a museum
 14. What are some examples of acceptable values for the parts of an HTTP response?
+* status: 200, body: OK
 15. What types of output do we want to test when we test our controllers?
+* http responses
```

Comment:
👍 We didn't teach controller testing this time around, so good job finding this!